src/Kfz24/Client/Aws/SqsClient.php

```diff
@@ -57,34 +22,84 @@ class SqsClient extends AbstractAwsClient
      */
     private $logger;
 
+    /**
+     * @var LargePayloadMessageExtension
+     */
+    private $largePayloadMessageExtension;
+
     /**
      * @param MessageValidator $validator
+     *
+     * @return SqsClient
      */
-    public function setValidator(MessageValidator $validator)
+    public function setValidator(MessageValidator $validator): SqsClient
     {
         $this->validator = $validator;
+
+        return $this;
     }
 
     /**
      * @param LoggerInterface $logger
+     *
+     * @return SqsClient
      */
-    public function setLogger(LoggerInterface $logger)
+    public function setLogger(LoggerInterface $logger): SqsClient
     {
         $this->logger = $logger;
+
+        return $this;
+    }
+
+    /**
+     * @param LargePayloadMessageExtension $largePayloadMessageExtension
+     *
+     * @return SqsClient
+     */
+    public function setLargePayloadMessageExtension(
+        LargePayloadMessageExtension $largePayloadMessageExtension
+    ): SqsClient {
+        $this->largePayloadMessageExtension = $largePayloadMessageExtension;
+
+        return $this;
     }
 
     /**
      * @param string|array $message
      *
      * @return Result
      *
+     * @throws \Exception
      */
     public function send($message)
     {
-        if (!is_array($message) || !array_key_exists('MessageBody', $message)) {
-            $message = ['MessageBody' => json_encode($message)];
+        if (is_array($message)) {
+            if (!array_key_exists('MessageBody', $message)) {
+                $message = ['MessageBody' => json_encode($message)];
+            }
+            else if (is_array($message['MessageBody']) || is_object($message['MessageBody'])) {
+                $message['MessageBody'] = json_encode($message['MessageBody']);
+            }
+        }
+
+        if (!is_array($message)) {
+            if (is_string($message) && $this->isJsonString($message)) {
+                $message = ['MessageBody' => $message];
+            }
+            else {
+                $message = ['MessageBody' => json_encode($message)];
```

Comment:
We should move `'MessageBody'` to a const. It's used quite often here. The same goes for `'ReceiptHandle'` and `'Body'` here in this class.